docs/manual/src/proc_macro/index.md

```diff
@@ -228,6 +228,56 @@ fn do_http_request() -> Result<(), MyApiError> {
 }
 ```
 
+## The `#[uniffi::callback_interface]` attribute
+
+`#[uniffi::callback_interface]` can be used to export a [callback interface](../udl/callback_interfaces.html) definition
+This allows the foreign bindings to implement the interface and pass an instance to the Rust code.
+
+```rust
+#[uniffi::callback_interface]
+pub trait Person {
+    fn name() -> String;
+    fn age() -> u32;
+}
+
+// Corresponding UDL:
+// callback interface Person {
+//   string name();
+//   u32 age();
+// }
+}
+```
+
+### Exception handling in callback interfaces
+
+Most languages allow arbitrary exceptions to be thrown, which presents issues for callback
+interfaces.  If a callback interface function returns a non-Result type, then any exception will
+result in a panic on the Rust side.
+
+To avoid panics, callback interfaces can use `Result<T, E>` types for all return values.  If the callback
+interface implementation throws the exception that corresponds to the `E` parameter, `Err(E)` will
+be returned to the Rust code.  However, in most languages it's still possible for the implementation
+to throw other exceptions.  To avoid panics in those cases, the error type must be wrapped
+with the `#[uniffi(handle_unknown_callback_error)]` attribute and
+`From<UnexpectedUniFFICallbackError>` must be implemented:
```

Comment:
Should this really be opt-in? I think an opt-out solution might be better.

Reply:
For callback interface errors I agree, but some (most?) errors are never used in callback interfaces and making them opt-out doesn't seem correct to me.

Reply:
Right, of course.